### PR TITLE
Update information on JSDoc

### PR DESCRIPTION
**Pull request by vforge**
Update some information on JSDoc based on what we're using in Mercury, so it's more consisted between our repositories.
Also, give background and more examples!

/cc @hakubo

### CODE REVIEW

File: JavaScript/CodingConventions.md [hunk 1]
```diff
@@ -517,6 +518,19 @@ function getRandomNumber(){
 var c = 3; /* A block comment after code */
 ```
 
+Good examples:
+```javascript
+// this variable is really needed
+var really_needed = 42;
```
Comment (rogatty) [inline JavaScript/CodingConventions.md]: `reallyNeeded`?

**Comment by rogatty** [comment]
LGTM
Let's ping @RafLeszczynski as he's a big fan of `@desc` ;)

**Comment by vforge** [comment]
According to http://usejsdoc.org/tags-description.html Example #1 - we can ommit `@description` tag when description is at the beginning of the file. It's also problematic with multi-paragraph descriptions, because you need to separate paragraphs with `\n` for better readability.

**Comment by RafLeszczynski** [comment]
:+1:

**Pull request merged**